src/components/MapComponent.vue

```diff
@@ -0,0 +1,262 @@
+      loadModules(['esri/Map', 'esri/views/MapView', 'esri/layers/FeatureLayer', 'esri/widgets/Locate'], { css: true })
+        .then(([ArcGISMap, MapView, FeatureLayer, Locate]) => {
+          const map = new ArcGISMap({
+            basemap: 'satellite',
+          });
+
+          // Main object with information about the map display
+          // https://developers.arcgis.com/javascript/latest/api-reference/esri-views-MapView.html
+          this.view = new MapView({
+            container: this.$el,
+            map,
+            center: [-71.0640, 42.3554], // Boston Common
+            zoom: 13.5,
+            popup: {
+              dockEnabled: true,
+              dockOptions: {
+              // Disables the dock button from the popup
+                buttonEnabled: false,
+                // Ignore the default sizes that trigger responsive docking
+                breakpoint: false,
+                // Set position of modal to bottom of the map
+                position: 'bottom-center',
+              },
+            },
+          });
+
+          // Converting layer objects to ArcGIS featureLayers so they can be
+          // interpreted by the MapView
+          const blockLayer = new FeatureLayer(this.blockLayerObject);
+          map.add(blockLayer);
+          const privateStreetLayer = new FeatureLayer(this.privateStreetLayerObject);
+          map.add(privateStreetLayer);
+
+          // Button that centers map over users current location
+          // https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Locate.html
+          const locate = new Locate({
+            view: this.view,
+            useHeadingEnabled: false,
+            goToOverride: (view, options) => view.goTo(options.target),
+          });
+          this.view.ui.add(locate, 'top-left');
+
+          // Opens a popup with the block information that corresponds with the activeBlockId
+          this.view.when(() => {
+            if (this.activeBlockId !== undefined) {
+            // Create a query where the ID equals the given ID
+              const query = blockLayer.createQuery();
+              query.where = `ID = ${this.activeBlockId}  AND RESERVED = 1`;
+              blockLayer.queryFeatures(query)
+                .then((response) => {
+                // ID is a key so there should only be one item in the
+                // features array that is returned
+                  const blockFeatures = response.features;
+                  // Sets what the popup should look like
+                  if (blockFeatures.length > 0) {
+                    blockFeatures[0].popupTemplate = this.blockPopupTemplate;
+                  }
+                  this.view.popup.open({
+                    features: blockFeatures,
+                  });
+                });
+            }
+          });
+
+          // Hides the completed block action when a block is completed
+          this.view.popup.watch('selectedFeature', (graphic) => {
```

Comment:
This replaces the need for the separate layer for the completed blocks. Disables the complete action if the selected block has a reservation status of 2